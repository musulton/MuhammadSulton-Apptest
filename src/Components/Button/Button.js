import * as React from 'react';
import {Text, TouchableOpacity} from 'react-native';

import styles from './Button.styles';

const Button = ({text, onPress}) => (
  <TouchableOpacity style={styles.button} onPress={onPress}>
    <Text style={styles.buttonText}>{text}</Text>
  </TouchableOpacity>
);

export default Button;
