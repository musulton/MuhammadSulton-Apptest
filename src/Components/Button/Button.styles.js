import {StyleSheet} from 'react-native';

import Colors from '../../Themes/Colors';

export default StyleSheet.create({
  button: {
    flex: 1,
    height: 30,
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonText: {
    flex: 1,
    fontSize: 16,
    fontWeight: '500',
    color: Colors.text,
  },
});
