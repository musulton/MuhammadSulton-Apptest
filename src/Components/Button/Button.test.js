import * as React from 'react';
import {render, cleanup} from '@testing-library/react-native';

import Button from './Button';

describe('Button', () => {
  afterEach(cleanup);

  it('should render button correctly', () => {
    const {toJSON} = render(<Button text={'example'} onPress={jest.fn} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
