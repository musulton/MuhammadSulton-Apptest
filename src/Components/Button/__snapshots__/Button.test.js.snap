// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": 30,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "flex": 1,
        "fontSize": 16,
        "fontWeight": "500",
      }
    }
  >
    example
  </Text>
</View>
`;
